packages/p2p-media-loader-hlsjs/src/engine.ts

```diff
@@ -10,16 +10,19 @@ import type { HlsConfig, Events } from "hls.js";
 import { FragmentLoaderBase } from "./fragment-loader";
 import { PlaylistLoaderBase } from "./playlist-loader";
 import { SegmentManager } from "./segment-mananger";
-import { Core, CoreEventMap } from "p2p-media-loader-core";
+import { Config, Core, CoreEventMap } from "p2p-media-loader-core";
 
+export type HlsEngineConfig = {
+  hlsjsConfig: Partial<HlsConfig>;
+} & Config;
 export class Engine {
   private readonly core: Core;
   private readonly segmentManager: SegmentManager;
   private hlsInstanceGetter?: () => Hls;
   private currentHlsInstance?: Hls;
 
-  constructor() {
-    this.core = new Core();
+  constructor(config: HlsEngineConfig) {
+    this.core = new Core(config);
     this.segmentManager = new SegmentManager(this.core);
   }
 
```

Comment:
I'm not sure where I can set config in hlsEngine. It seems like we create config here: 
```

public getConfig(): Partial<HlsConfig> {
    return {
      fLoader: this.createFragmentLoaderClass(),
      pLoader: this.createPlaylistLoaderClass(),
    };
  }
```
So I thought about something like this to set config:

```
private createFragmentLoaderClass() {
    const core = this.core;
    // eslint-disable-next-line @typescript-eslint/no-this-alias
    const engine: Engine = this;
    const config = this.config;

    return class FragmentLoader extends FragmentLoaderBase {
      constructor() {
        super(config.hlsjsConfig as HlsConfig, core); // 'as' is used due to Partial type in config
      }

      static getEngine(): Engine {
        return engine;
      }
    };
  }

  private createPlaylistLoaderClass() {
    // eslint-disable-next-line @typescript-eslint/no-this-alias
    const engine: Engine = this;
    const config = this.config;
    return class PlaylistLoader extends PlaylistLoaderBase {
      constructor() {
        super(config.hlsjsConfig as HlsConfig);
        engine.initHlsEvents();
      }
    };
  }
```

Is this thought right?